docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,726 @@
+The type `$null` contains only `null.null` (which is equivalent to `null`).
+The nominal Ion types are `$blob`, `$bool`, `$clob`, `$decimal`, `$float`, `$int`, `$null`, `$string`, `$symbol`, `$timestamp`, `$list`, `$sexp`, and `$struct`.
+
+Ion Schema adds one more nominal type, `document`, which is a stream of top-level Ion values.
+Ion Schema implementations may represent this stream as a list, sequence, array, or other similar data structure depending on what is idiomatic for the language in which it is implemented.
+
+In addition, Ion Schema provides the following structural types:
+
+- `$any`: the _universal type_ that includes every value. It is the union of all nominal types, including `$null`.
+- `$lob`: union of `$blob` or `$clob`
+- `$number`: union of `$decimal`, `$float`, or `$int`
+- `$text`: union of `$string` or `$symbol`
+
+- `blob`, `bool`, `clob`, `decimal`, `float`, `int`, `string`, `symbol`, `timestamp`, `list`, `sexp`, `struct` correspond to the Ion types, except that they do not allow null values.
+- `any`: represents any value that is not a null value. It is the union of `blob`, `bool`, `clob`, `decimal`, `float`, `int`, `string`, `symbol`, `timestamp`, `list`, `sexp`, `struct`, and `document`.
+- `lob`: union of `blob` or `clob`
+- `number`: union of `decimal`, `float`, or `int`
+- `text`: union of `string` or `symbol`
+- `nothing`: the [_empty type_](https://en.wikipedia.org/wiki/Empty_type); it has no valid value
+
+{% include note.html type="tip" content="
+For the built-in types, the presence of a leading `$` sigil generally indicates that the type includes the respective strongly-typed null values.
+The universal type `$any` is an exception since it also includes the untyped null.
+" %}
+
+{% include note.html type="note" content="
+You may have noticed that there is a `$null` type, but there is no corresponding `null` type (as there is with `$int` and `int`).
+That is because a non-null equivalent of the type `$null` is an empty type.
+The non-null equivalent of `$null` is actually `nothing`.
+" %}
+
+## Type Definitions
+
+{% include grammar-element.md productions="named_type_definition,inline_type_definition,variably_occurring_type_reference" %}
+
+A type consists of a collection of zero or more constraints.
+The set of values which belong to a type is the intersection of the values that satisfy each constraint.
+In order for a value to be a valid instance of a type, the value must not violate any of the type's constraints.
+A type definition with no constraints is equivalent to `$any`.
+
+### Declaring a Type Name
+
+A _named type definition_ is a special case of type definition.
+A named type definition must be a top-level value in the schema document that is annotated with `type`. 
+A named type definition must contain the field `name`, where the value is an Ion symbol that is to be the name of the type.
```

Comment:
It's mostly just from historical precedent—it keeps things a little simpler for implementations if the rule here is the same for Ion Schema 1.0 and 2.0. Requiring only a symbol means we don't have to explain whether `"foo"`  and `'foo'` are considered as equivalent or distinct type names.

Do you know of any reasons to allow `name` to be text instead of just a symbol?

Reply:
> Do you know of any reasons to allow name to be text instead of just a symbol?

Not really, just that intuitively it's not the name's *type* that matters, just its content. There isn't a good enough reason at this point to diverge from the precedent.